ceph-salt-formula/salt/ceph-salt/cephtools.sls

```diff
@@ -35,11 +35,14 @@ have cephadm check the host:
 
 {{ macros.end_step('Run "cephadm check-host"') }}
 
+{% set image = pillar['ceph-salt']['container']['images']['ceph'] %}
+
 {{ macros.begin_step('Download ceph container image') }}
 download ceph container image:
   cmd.run:
     - name: |
-        cephadm --image {{ pillar['ceph-salt']['container']['images']['ceph'] }} pull
+        cephadm --image {{ image }} pull
+    - unless: podman image exists {{ image }}
```

Comment:
This will change the expected behavior of a pull. If you have set the image to eg `registry.opensuse.org/opensuse/ceph/ceph:latest` I would expect that it takes the **latest** image available on the registry. With this code change, it will never check again if there is a newer version of the image available.
Or am I missing something here?

Reply:
I'd go the kubernetes route and discourage the usage of `:latest` altogether.

Reply:
Or if we want to change the default `pull` behavior to NOT pull the latest image if we already have a local one, then maybe this can be implemented in `cephadm pull` directly instead of `ceph-salt`? 

Something like:
```
# cephadm pull 
Local image already exists.
# cephadm pull --force
Pulling image....
``` 

Thoughts?

Reply:
> This will change the expected behavior of a pull. If you have set the image to eg `registry.opensuse.org/opensuse/ceph/ceph:latest` I would expect that it takes the **latest** image available on the registry. With this code change, it will never check again if there is a newer version of the image available.
> Or am I missing something here?

I think you are right if ":latest" is being used, and if so, you definitely want the images to be updated even if you have already downloaded them. Requiring a "--force" doesn't seem to be intuitive or helpful in this case.

Reply:
> I'd go the kubernetes route and discourage the usage of `:latest` altogether.

`latest` was just an example. It could also be `15` which would always pull the latest minor/patch releases for the major version `15`